### PR TITLE
32

### PR DESCRIPTION
**Pull request by cichan02**

### CODE REVIEW

**Comment by cichan02** [comment]
@h1alexbel Pay attention on src/main/java/com/taskviewer/api/web/rq/RqTaskSearchCriteria.java

File: src/main/java/com/taskviewer/api/web/rq/RqTaskSearchCriteria.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
+package com.taskviewer.api.web.rq;
+
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.NotNull;
+
+public record RqTaskSearchCriteria(String username, String status, Integer priority) {
+
+	@Contract(value = " -> new", pure = true)
```
Comment (h1alexbel) [inline src/main/java/com/taskviewer/api/web/rq/RqTaskSearchCriteria.java:8]: @cichan02 what is it?

**Review by h1alexbel — CHANGES_REQUESTED** [review]

File: src/main/java/com/taskviewer/api/web/rq/RqTaskSearchCriteria.java (hunk 1 above)
Comment (cichan02) [reply src/main/java/com/taskviewer/api/web/rq/RqTaskSearchCriteria.java:8]: Thanks to this annotation, we can define a contract that our methods must match. JetBrains, the brand behind [IntelliJ IDEA](https://www.baeldung.com/intellij-basics), introduced the annotation. It allows the text editor to directly troubleshoot potential problems with the calling methods in our code.

The @Contract annotation has two attributes: value and pure. The value attribute contains clauses describing the relationship between the methods' inputs and output. 

A contract is a set of causality clauses of the form: “A -> B”. This means that providing A to the method will always give B as a result. For instance, “_ -> null” means that the method returns null for any input value.

Comment (cichan02) [reply src/main/java/com/taskviewer/api/web/rq/RqTaskSearchCriteria.java:8]: In my case that means, that method returns new for any input

File: src/main/java/com/taskviewer/api/postgres/PgTasks.java [hunk 2]
```diff
@@ -2,57 +2,178 @@
...
+	protected static final String FIND_WITH_STATUS = """
+		SELECT t.id AS task_id,
+			   t.title AS title,
+			   t.about AS about,
+			   t.status AS status,
+			   t.priority AS priority,
+			   t.estimate AS estimate,
+			   t.tracked AS tracked,
+			   t.created as task_created,
+
```
[the remark this answers is not in the document]
Comment (cichan02) [reply src/main/java/com/taskviewer/api/postgres/PgTasks.java]: Removed spaces in sql

**Comment by cichan02** [comment]
Rebase to master to avoid merge conflicts in the future.

**Review by h1alexbel — APPROVED** [review]

File: src/main/java/com/taskviewer/api/web/rq/RqTaskSearchCriteria.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
+package com.taskviewer.api.web.rq;
+
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.NotNull;
+
+public record RqTaskSearchCriteria(String username, String status, Integer priority) {
```
[the remark this answers is not in the document]
Comment (cichan02) [reply src/main/java/com/taskviewer/api/web/rq/RqTaskSearchCriteria.java:6]: ok, will be added in evening after I come to office.

[the remark this answers is not in the document]
Comment (cichan02) [reply src/main/java/com/taskviewer/api/web/rq/RqTaskSearchCriteria.java:6]: If it not critical

[the remark this answers is not in the document]
Comment (cichan02) [reply src/main/java/com/taskviewer/api/web/rq/RqTaskSearchCriteria.java]: @h1alexbel added

**Review by h1alexbel — APPROVED** [review]

**Comment by cichan02** [comment]
@h1alexbel fix merge conflict

**Review by h1alexbel — APPROVED** [review]

**Comment by h1alexbel** [comment]
@cichan02 thanks!

**Pull request merged**